File: chapter6/salesitemservice/src/common/authorizer/FakeAuthorizer.ts
import { Authorizer } from './Authorizer';
import { Injectable } from '@nestjs/common';

@Injectable()
class FakeAuthorizer implements Authorizer {
  authorizeIfUserHasOneOfRoles(roles: string[]): void {
    console.log('Authorized user with one of the roles: ' + roles.toString());
  }
}

export const authorizer = new FakeAuthorizer();

File: chapter2/orderservice/src/main/java/com/example/orderservice/common/authorizer/FakeAuthorizer.java
package com.example.orderservice.common.authorizer;

import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class FakeAuthorizer implements Authorizer {
  @Override
  public void authorize(final Optional<String> maybeAuthHeader) {

  }

  @Override
  public void authorizeForSelf(final Optional<String> maybeUserId, final Optional<String> maybeAuthHeader) {

  }

  @Override
  public void authorizeIfUserHasOneOfRoles(final List<String> allowedRoles, final Optional<String> maybeAuthHeader) {
    System.out.println("Authorized for role(s) " + allowedRoles + ", with auth header: " + maybeAuthHeader.orElse(""));
  }

  @Override
  public String getUserId(final Optional<String> maybeAuthHeader) {
    return "";
  }
}
